### ISSUE TITLE
Support setting MongoDB readPreference

### ISSUE THREAD

**Issue by tchwpkgorg** [metabase/metabase#4491]
Currently, there is no way to provide readPreference for mongo.

For example - for connectors which support it - readPreference which would tell mongo to use the physically nearest servers would look like:

mongodb://10.11.12.13/?readPreference=nearest

**Comment by camsaul** [comment]
We could probably add a field for additional connection options so you could just enter

```
readPreference=nearest
```

or other options as needed. That might solve #1374 as well

**Comment by camsaul** [comment]
actually it might as easy as I was hoping because we're setting up the Mongo connection using the Mongo Java library's [MongoClientOptions.Builder](http://api.mongodb.com/java/current/com/mongodb/MongoClientOptions.Builder.html) rather than just passing in a URL with the various options tacked on at the end 

So it wouldn't be as simple as just taking on `?readPreference=nearest` to a connection string, it would be more like


```clojure
(.readPreference builder (ReadPreference/nearest))
```

**Comment by camsaul** [comment]
nvm I totally found a way to hack this.

You can pass use a Mongo connection string to construct a MongoClientURI and then call .getOptions on it to get an instance of `MongoClientOptions` with the parameters set appropriately

```clojure
(.getOptions (com.mongodb.MongoClientURI. "mongodb://localhost/?readPreference=nearest"))

;; -> #com.mongodb.MongoClientOptions{description='null', readPreference=nearest, ...}
```

**Comment by camsaul** [comment]
Implemented by #4971
